src/doc/performance-tuning.md

```diff
@@ -257,6 +439,71 @@ root@c819e0f106c4:/# taskset -cp $$
 pid 1's current affinity list: 0-15
 ```
 
+## Application-specific performance notes
+
+### lwAFTR
+
+#### Avoid fragmentation
+
+Fragment reassembly is a costly operation.  Make sure that network
+MTUs are set such that fragmentation is rare.
+
+#### Ingress and egress filtering
 
+Simply enabling ingress and/or egress filtering has a cost.  Enabling
+all filters adds 4 apps to the Snabb graph, and there is a cost for
+every additional Snabb app.  In our tests, while a normal run can do
+10 Gbps over two interfaces in full duplex, enabling filters drops
+that to 8.2 Gbps before dropping packets.  However we have not found
+that the performance depends much on the size of the filter, as the
+filter's branches are well-biased.
 
+## Programming notes
+
+Here are a couple of notes for people writing their first Snabb data
+plane.
+
+### Enable engine.busywait
+
+Defined in src/core/app.lua and enabled before calling engine.main()
+via
+
+```
+engine.busywait = true
+engine.main(...)
+```
+
+If true then the engine will poll for new data in a tight loop (100%
+CPU) instead of sleeping according to the Hz setting. This will reduce
+overall packet latency and increase throughput at the cost of
+utilizing the CPU hosting Snabb at 100%.
+
```

Comment:
Want to comment on what we've empirically observed with enabling/disabling busywait?